pkg/orchestrator/provider/aws/scanestimation/estimate.go

```diff
@@ -78,17 +78,84 @@ const (
+// The times correspond to the scan size values {0.01, 2.5, 8.1}.
 // nolint:gomnd
-var familyScanDurationPerGBMap = map[familiestypes.FamilyType]time.Duration{
-	familiestypes.SBOM:             time.Duration(4.5 * float64(time.Second)),
-	familiestypes.Vulnerabilities:  1 * time.Second, // TODO check time with no sbom scan
-	familiestypes.Secrets:          300 * time.Second,
-	familiestypes.Exploits:         0,
-	familiestypes.Rootkits:         0,
-	familiestypes.Misconfiguration: 1 * time.Second,
-	familiestypes.Malware:          360 * time.Second,
+var familyScanDurationsMap = map[familiestypes.FamilyType][]float64{
+	familiestypes.SBOM:             {0.01, 11, 37},
+	familiestypes.Vulnerabilities:  {0.01, 1, 11}, // TODO check time with no sbom scan
+	familiestypes.Secrets:          {0.01, 720, 1320},
+	familiestypes.Exploits:         {0, 0, 0},
+	familiestypes.Rootkits:         {0, 0, 0},
+	familiestypes.Misconfiguration: {0.01, 4, 5},
+	familiestypes.Malware:          {0.01, 840, 2460},
+}
+
+// familyCoefficients populated during init in order to extrapolate the scan duration from scan size per family.
+// The extrapolation formula is y = a * ln(x) + b.
+var familyCoefficients = make(map[familiestypes.FamilyType][]float64, len(families))
+
+func init() {
+	if err := initFamilyScanDurationCoefficients(); err != nil {
+		logrus.Errorf("Failed to init family scan duration coefficients: %v", err)
+	}
+}
+
+func initFamilyScanDurationCoefficients() error {
+	var err error
+	var xCoefficient, yCoefficient float64
+	// xData represents the used memory size in GB.
+	xData := []float64{0.01, 2.5, 8.1}
+
+	for _, family := range families {
+		// yData represents the times in seconds that the family scan took per the xData values.
+		yData := familyScanDurationsMap[family]
+		xCoefficient, yCoefficient, err = logarithmicFit(xData, yData)
+		if err != nil {
+			return fmt.Errorf("failed to calculate least squares fitting: %w", err)
+		}
+		familyCoefficients[family] = []float64{xCoefficient, yCoefficient}
+	}
+
+	return nil
+}
+
+// logarithmicFit performs least squares fitting for the logarithmic model y = a * ln(x) + b.
+func logarithmicFit(xData, yData []float64) (float64, float64, error) {
+	n := len(xData)
+	if n != len(yData) || n == 0 {
+		return 0, 0, fmt.Errorf("input data is invalid. xData: %v, yData: %v", xData, yData)
+	}
+
+	lnXData := make([]float64, n)
+
+	for i := 0; i < n; i++ {
+		if xData[i] <= 0 {
+			return 0, 0, fmt.Errorf("input data contains non-positive x values: %v", xData)
+		}
+		lnXData[i] = math.Log(xData[i])
+	}
+
+	// Calculate the coefficients of the linear regression model y = a * ln(x) + b
+	var sumLnX, sumY, sumLnX2, sumLnXY float64
+	for i := 0; i < n; i++ {
+		sumLnX += lnXData[i]
+		sumY += yData[i]
+		sumLnX2 += lnXData[i] * lnXData[i]
+		sumLnXY += lnXData[i] * yData[i]
+	}
+
+	a := ((float64(n))*sumLnXY - sumLnX*sumY) / ((float64(n))*sumLnX2 - sumLnX*sumLnX)
```

Comment:
I think that worth adding a non zero check for ((float64(n))*sumLnX2 - sumLnX*sumLnX) to be on the safe side